platform/ui/src/components/InputDoubleRange/InputDoubleRange.tsx

```diff
@@ -0,0 +1,219 @@
+    setSelectedThumb(null);
+  };
+
+  const handleGlobalMouseUp = () => {
+    // Remove global mouse event listeners
+    window.removeEventListener('mousemove', handleMouseMove);
+    window.removeEventListener('mouseup', handleGlobalMouseUp);
+    setSelectedThumb(null); // Stop the dragging
+  };
+
+  const handleMouseDown = e => {
+    const rect = e.currentTarget.getBoundingClientRect(); // Use currentTarget instead of target
+    const x = e.clientX - rect.left;
+    const percentageClicked = (x / rect.width) * 100;
+
+    // Calculate the distances from the clicked point to both thumbs' positions
+    const distanceToStartThumb = Math.abs(percentageClicked - percentageStart);
+    const distanceToEndThumb = Math.abs(percentageClicked - percentageEnd);
+
+    // Check if the clicked point is within a threshold distance to either thumb
+    if (distanceToStartThumb < 10) {
+      setSelectedThumb(0);
+    } else if (distanceToEndThumb < 10) {
+      setSelectedThumb(1);
+    }
+
+    window.addEventListener('mousemove', handleMouseMove);
+    window.addEventListener('mouseup', handleGlobalMouseUp);
+  };
+
+  const handleMouseMove = e => {
+    if (selectedThumb !== null) {
+      const rect = e.currentTarget.getBoundingClientRect();
+      const x = e.clientX - rect.left;
+      const newValue =
+        Math.round(((x / rect.width) * (maxValue - minValue) + minValue) / step) * step;
+
+      // Make sure newValue is within [minValue, maxValue]
+      const clampedValue = Math.min(Math.max(newValue, minValue), maxValue);
+
+      // Ensure that left and right thumbs don't switch positions
+      if (selectedThumb === 0 && clampedValue >= rangeValue[1]) {
+        return;
+      }
+      if (selectedThumb === 1 && clampedValue <= rangeValue[0]) {
+        return;
+      }
+
+      // Update the correct values in the rangeValue array
+      const updatedRangeValue = [...rangeValue];
+      updatedRangeValue[selectedThumb] = clampedValue;
+      setRangeValue(updatedRangeValue);
+
+      // No need to call onChange here, since we're updating local state+
+      onChange(updatedRangeValue);
+
+      // Update the thumb position
+      const percentage = Math.round(((clampedValue - minValue) / (maxValue - minValue)) * 100);
+      if (selectedThumb === 0) {
+        setPercentageStart(percentage);
+      } else {
+        setPercentageEnd(percentage);
+      }
+    }
+  };
+
+  // Calculate the range values percentages for gradient background
+  const rangeValuePercentageStart = ((rangeValue[0] - minValue) / (maxValue - minValue)) * 100;
+  const rangeValuePercentageEnd = ((rangeValue[1] - minValue) / (maxValue - minValue)) * 100;
+
+  return (
+    <div className={`flex items-center space-x-2${containerClassName ?? ''}`}>
+      {showLabel && LabelOrEditableNumber(rangeValue[0], 0)}
+      <div
+        className="relative flex h-10 w-full items-center"
+        onMouseDown={handleMouseDown}
+        onMouseMove={handleMouseMove}
```

Comment:
I don't think we need to add the `onMouseMove` and `onMouseUp` here since they are being added on the window.

Reply:
No they should be there, they decide which thumb is dragged and move it

Reply:
Ok. I find it odd that we need both these AND the listeners on the window, but perhaps I am missing something. In fact, I think maybe the mouse move listener on the window has a stale/null selected thumb and might be doing nothing. Maybe only the mouse up listener on the window is useful. Anyway please have a another look.